### PR TITLE
Adapt jacoco to pick-up kotlin classes

### PR DESCRIPTION
**Pull request by LukasPaczos**
The old script was prepared to work with Java files only. This changes the plugin to easily pick up Kotlin as well.

@Guardiola31337 do you know whether this requires any additional changes to the codecov script? Especially for the legacy codebase.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -119,7 +119,7 @@ commands:
       - run:
           name: Run unit-test, generate Jacoco test report and Post code coverage report to Codecov.io
           command: |
-            ./gradlew testDebugUnitTestCoverage
+            ./gradlew jacocoTestReleaseUnitTestReport
```
Comment (LukasPaczos) [inline circle.yml]: I opted to use release build for codecov base instead debug, any issues this might cause?

File: gradle/jacoco.gradle [hunk 2]
```diff
@@ -1,38 +1,9 @@
...
-    }
-    build.dependsOn "${coverageTaskName}"
-  }
+jacoco {
+  toolVersion = "0.8.3"
 }
+
+tasks.withType(Test) {
+  jacoco.includeNoLocationClasses = true
+}
```
Comment (Guardiola31337) [inline gradle/jacoco.gradle]: Do we need this 🤔? It's also included in https://github.com/mapbox/mapbox-navigation-android/blob/fad1cd4de295399b999dc5009b66cad2e567b01a/libandroid-navigation/build.gradle#L37-L39

File: circle.yml (hunk 1 above)
Comment (Guardiola31337) [reply circle.yml]: Yeah 👀 https://github.com/codecov/example-android#instrumentation-tests-coverage-reports we should enable `testCoverageEnabled true` for the `release` `buildType` in https://github.com/mapbox/mapbox-navigation-android/blob/fad1cd4de295399b999dc5009b66cad2e567b01a/libandroid-navigation/build.gradle#L26-L30 The only thing that I'm unsure about is if Codecov will retrieve the reports seamlessly or if we need to set the release path explicitly, although I imagine that it'll 👀 to `build/reports/jacoco` and get all the reports for the different variants so I believe we should be good.

Comment (LukasPaczos) [reply circle.yml]: Reverted back to debug and used the flag in only one place.

**Comment by Guardiola31337** [comment]
Interesting CI (`D8`) failure 🤔

```
Caused by: com.android.tools.r8.utils.AbortException: Error: /root/code/examples/build/intermediates/dex/debug/mergeDexDebug/out/classes2.dex, Cannot allocate memory
Stack trace:
java.io.IOException: Cannot allocate memory
	at sun.nio.ch.FileDispatcherImpl.write0(Native Method)
	at sun.nio.ch.FileDispatcherImpl.write(FileDispatcherImpl.java:60)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:211)
	at java.nio.channels.Channels.writeFullyImpl(Channels.java:78)
	at java.nio.channels.Channels.writeFully(Channels.java:101)
	at java.nio.channels.Channels.access$000(Channels.java:61)
	at java.nio.channels.Channels$1.write(Channels.java:174)
	at com.android.tools.r8.utils.FileUtils.a(:37)
	at com.android.tools.r8.utils.s.a(:21)
	at com.android.tools.r8.DexIndexedConsumer$DirectoryConsumer.accept(:7)
	at com.android.tools.r8.dex.ApplicationWriter.a(:56)
	at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

	at com.android.tools.r8.utils.Reporter.a(:21)
	at com.android.tools.r8.dex.ApplicationWriter.write(:90)
	at com.android.tools.r8.D8.d(:44)
	at com.android.tools.r8.D8.b(:1)
	at com.android.tools.r8.utils.t.a(:23)
	... 32 more


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':examples:mergeDexDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
```

cc @LukasPaczos

**Comment by LukasPaczos** [comment]
Maybe a CI failure. I restarted and we'll see.

**Pull request merged**